File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        vector<int> res;
        for(int n:nums){
            mp[n]++;
        }
        vector<pair<int,int>>arr;
        for(const auto& p:mp){
            arr.push_back({p.second,p.first});
        }
        sort(arr.rbegin(),arr.rend());

        for(int i=0;i<k;i++){
            res.push_back(arr[i].second);
        }
    return res;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> freqMap=new HashMap<>();
        for(int n:nums){
            freqMap.put(n,freqMap.getOrDefault(n,0)+1);
        }

        List<Integer>[] arr= new List[nums.length+1];

        for(int key:freqMap.keySet()){
            int freq=freqMap.get(key);
            if(arr[freq]==null){
                arr[freq]=new ArrayList<>();
            }
            arr[freq].add(key);
        }

        List<Integer> res=new ArrayList<>();
        for(int i=arr.length-1;i>=0;i--){
            if(arr[i]!=null && res.size()<k){
                res.addAll(arr[i]);
            }
        }
        int[] result=new int[k];
        for(int i=0;i<k;i++){
            result[i]=res.get(i);
        }
    return result;
    }
}